### PR TITLE
Created `Signing.verify(message:hasValidSignature:with:)`

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes [CSDK-633].
This adds `Signing.verify(message:nonce:hasValidSignature:with:)`.

Thanks to @tonidero for helping with the signature part (#2253).

[CSDK-633]: https://revenuecats.atlassian.net/browse/CSDK-633?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by NachoSoto** [comment]
This is pretty much ready.

**Review by tonidero — APPROVED** [review]
Looks great!

File: Tests/UnitTests/Misc/SigningTests.swift [hunk 1]
```diff
@@ -58,4 +61,69 @@ class SigningTests: TestCase {
...
+
+    func testVerifySignatureLogsWarningWhenFail() throws {
+        let logger = TestLogHandler()
+
+        _ = Signing.verify(message: "Hello World".asData,
+                           nonce: "nonce".asData,
+                           hasValidSignature: "invalid signature".asData.base64EncodedString(),
+                           with: try Signing.loadPublicKey())
+
+        logger.verifyMessageWasLogged("Signature failed validation", level: .warn)
```
Comment (aboedo) [inline Tests/UnitTests/Misc/SigningTests.swift:94]: I love that we have tests for logging stuff now

**Review by aboedo — COMMENTED** [review]
awesome

File: Tests/UnitTests/Misc/SigningTests.swift (hunk 1 above)
Comment (NachoSoto) [reply Tests/UnitTests/Misc/SigningTests.swift:94]: 💯

**Pull request merged**